sale_order_line_description/models/sale.py

```diff
@@ -19,36 +19,31 @@
 #
 #
 
-from openerp.osv import orm
+from openerp import models, api
 
 
-class SaleOrderLine(orm.Model):
+class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
+    @api.multi
     def product_id_change(
-        self, cr, uid, ids, pricelist, product, qty=0,
-        uom=False, qty_uos=0, uos=False, name='', partner_id=False,
-        lang=False, update_tax=True, date_order=False, packaging=False,
-        fiscal_position=False, flag=False, context=None
-    ):
+        self, pricelist, product, qty=0, uom=False,
+        qty_uos=0, uos=False,
+        name='', partner_id=False, lang=False, update_tax=True,
+        date_order=False, packaging=False, fiscal_position=False,
+            flag=False):
         res = super(SaleOrderLine, self).product_id_change(
-            cr, uid, ids, pricelist, product, qty=qty, uom=uom,
-            qty_uos=qty_uos, uos=uos, name='', partner_id=partner_id,
-            lang=lang, update_tax=update_tax, date_order=date_order,
-            packaging=packaging, fiscal_position=fiscal_position,
-            flag=flag, context=context
-        )
+            pricelist, product, qty, uom, qty_uos, uos, name,
+            partner_id, lang, update_tax, date_order, packaging,
+            fiscal_position, flag)
         if product:
+            product_obj = self.env['product.product']
             if self.user_has_groups(
-                    cr, uid,
                     'sale_order_line_description.'
-                    'group_use_product_description_per_so_line',
-                    context=context
-            ):
-                product = self.pool['product.product'].browse(
-                    cr, uid, product, context=context)
-                if product.description_sale:
+                    'group_use_product_description_per_so_line'):
+                product_reg = product_obj.browse(product)
```

Comment:
I prefered to maintain product as it comes from product_id_change with only id, and make a new var for its registry, but as you wish.